### PR TITLE
[BUGFIX release-1-13] Fix {{input}} helper on IE8.

### LINKED ISSUE #11553: IE8: {{input type=""}} returns an error
Using IE8 and Ember.js 1.13.2

When using `{{input type="checkbox"}}` IE8 shows a "this command is not supported" error. I guess this is the result of the type being set after the input element is rendered which is not possible in IE8.

http://stackoverflow.com/questions/18240063/ie8-throws-this-command-is-not-supported-error-when-using-knockout-for-data-bi

### LINKED ISSUE #12295: Radio buttons render as a text input in IE8
Here is a very tiny example: http://emberjs.jsbin.com/kiyesi 

`<input type="radio">` of course renders fine in IE8 but rendering the same minimal example dynamically via a component results in a text input instead:

``` js
App.RadioCheckComponent = Ember.Component.extend({
  tagName: 'input',
  type: 'radio',
  attributeBindings: ['type'],
});
```

I just upgraded from Ember v1.12.x and the problem did not exist there.

### PR DESCRIPTION
**Pull request by rwjblue**
Closes #12351.
Fixes #11553.
Fixes #12295.
Closes tildeio/htmlbars#380.

### CODE REVIEW

**Comment by rwjblue** [comment]
I have pushed builds for this to https://github.com/rwjblue/ember/tree/ie8-input-type to ease testing.

**Comment by rwjblue** [comment]
@gauthierm / @xcskier56 - Please test the assets provided above, and confirm that the various permutations work properly.

**Comment by gauthierm** [comment]
@rwjblue I tested the build provided above and it works for both attributes in the template and for attributes specified in the component definition. You refactor looks great.

File: packages/ember-views/lib/system/build-component-template.js [hunk 1]
```diff
@@ -135,17 +160,32 @@ function normalizeComponentAttributes(component, isAngleBracket, attrs) {
       if (colonIndex !== -1) {
         var attrProperty = attr.substring(0, colonIndex);
         attrName = attr.substring(colonIndex + 1);
-        expression = ['get', 'view.' + attrProperty];
+
+        if (attrName === 'type' && hardCodeType) {
+          expression = component.get(attrProperty);
```
Comment (mixonic) [inline packages/ember-views/lib/system/build-component-template.js]: Setting  value to `expression` here works? I am a bit flummoxed by that.

Comment (rwjblue) [inline packages/ember-views/lib/system/build-component-template.js]: @mixonic and I discussed this in chat.  When set to a string HTMLBars `manualElement` calls `dom.setAttribute` **before** appending to the fragment, any value other than a string here would result in the element being created and appended to a fragment, *then* the attributes would be set which would trigger the error on IE8.

Relevant code paths:

* https://github.com/tildeio/htmlbars/blob/master/packages/htmlbars-runtime/lib/render.js#L86-L87
* https://github.com/tildeio/htmlbars/blob/master/packages/htmlbars-runtime/lib/render.js#L108-L109

**Comment by jerel** [comment]
@rwjblue per my testing this does not yet fix #12295. Here is a jsbin using the code from https://github.com/rwjblue/ember/tree/ie8-input-type: http://output.jsbin.com/hiriribulu/1 Fire up Edge in IE8 mode or use IE8 and it's still a text input.

**Comment by gauthierm** [comment]
@jerel Your jsbin in IE8 on Windows XP works properly. Edge in emulates IE8 mode does not do a good job of emulating this bug. The type attribute is mutable in emulated IE8 mode in Edge.

**Comment by gauthierm** [comment]
@jerel fwiw, you can get IE8 VMs for testing from modern.ie.

**Comment by jerel** [comment]
@gauthierm well, interesting. I can confirm that it works properly in genuine IE8 on WIndows 7. This fix feels a little shaky since it doesn't work in Edge: I do IE8 debugging in Edge when possible because the tools are so much better and now all radio inputs are broken there but not in IE8.

**Comment by gauthierm** [comment]
Unfortunately Edge doesn't emulate IE8 properly and can't be relied on for IE8 JS testing or debugging. This fix is correct in that is does feature detection to see if the type attribute is mutable. In Edge emulated IE8, the type attribute is mutable. In real IE8 the type attribute is not mutable.

**Comment by rwjblue** [comment]
> This fix feels a little shaky since it doesn't work in Edge

How can the fact that Edge is not properly emulating IE8 be this patches fault?

**Pull request merged**